### PR TITLE
move zarr, xarray, dask from examples to tests

### LINKED ISSUE #179: Examples should only have dependencies that are in the requirements
## 📚 Documentation
I think the scripts in the examples directory should be guaranteed to run immediately after a fresh installation (i.e., they should only have dependencies that are in the requirements). It's a little frustrating to do the installation, go to try out napari and then get a `ModuleNotFoundError`. What do you all think?

### PR DESCRIPTION
**Pull request by sofroniewn**
# Description
This PR closes #179 by moving our examples of zarr, xarray, and dask to our tests. I think that as we don't explicitly depend on these packages its more confusing to have them in our examples as they won't run when a new-commer installs napari. We now though do test for integration.

There are two problems that I noted, one is documented in #554 around autocreation of pyramids sometimes being undesirable. Another is that right now if using a zarr array without wrapping in dask you must sent `contrast_limits` explicitly otherwise things will crash.

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)

# How has this been tested?

- [x] add `test_numpy_like.py`

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: napari/tests/test_numpy_like.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
+import numpy as np
```
Comment (jni) [inline napari/tests/test_numpy_like.py:1]: Should this file not include a `pytest.importorskip`?

File: napari/util/misc.py [hunk 2]
```diff
@@ -246,12 +246,17 @@ def calc_data_range(data):
...
-    min = data.min()
-    max = data.max()
-    if min == max:
-        min = 0
-        max = 1
-    return [float(min), float(max)]
+    if np.prod(data.shape) > 1e6:
+        reduced_data = np.random.choice(data.flatten(), size=1e6)
+    else:
+        reduced_data = data
```
Comment (jni) [inline napari/util/misc.py:260]: Not in love with this change... Accessing 1M random points from a dask array could be extremely slow depending on backing (e.g. remote data). I still prefer dtype, but if you don't want that, I suggest taking e.g. the middle plane.

**Review by jni — COMMENTED** [review]
@sofroniewn why move instead of copy? I think they are valuable as examples.

I also had some deeper concerns about the new additions... What do you think?

File: napari/util/misc.py (hunk 2 above)
Comment (sofroniewn) [reply napari/util/misc.py:260]: yup, it's a concern for me too, if each of those is in a different chunk .... I just wasnt quite sure what to do. How about top, middle, and bottom planes? Seems like an ok compromise for now!!

**Comment by sofroniewn** [comment]
The move instead of copy was because the examples won't run unless you have those packages installed. Maybe we need to add a try and catch with an error message saying "Dask not installed, in order to run dask example you must install dask with pip install dask" or something like that?

I can work on making these two changes and merge if you approve :-)

**Review by jni — APPROVED** [review]
Yep, I like all your proposed solutions, go for it! =)

File: napari/tests/test_numpy_like.py (hunk 1 above)
Comment (sofroniewn) [reply napari/tests/test_numpy_like.py:1]: These packages are listed in our test dependencies so I don't think we need the `importorskip`, or am I mistaken?

**Comment by sofroniewn** [comment]
I've also made an improvement to the multichannel api here (maybe should been it's own PR, sorry) that makes the keyword arg `axis` instead of `channel` and removes some duplicate code

i will merge this once the tests pass

**Pull request merged**